lib/messages.js

```diff
@@ -0,0 +1,40 @@
+// @ts-nocheck
+/* eslint-disable */
+import _m0 from "protobufjs/minimal.js";
+function createBaseExtension() {
+    return { authCoreKeys: [], wantCoreKeys: [] };
+}
+export var Extension = {
```

Comment:
seems like this would this be in the core-manager directory.

it could use a more detailed name. something like `CoreExchangeExtension`?

Reply:
I was thinking we would use a single extension for all messages we send (e.g. for sharing sync state), rather than register a separate extension for each one. This is inspired by how [hyperbee uses extensions](https://github.com/holepunchto/hyperbee/blob/master/schema.proto). For additional messages we would them as optional messages attached to the extension class.

That said, maybe `MapeoExtension` is a better name? And since I think the core manager would likely manage all extensions, I think it makes sense to move to there.